hbase/src/test/java/org/apache/metamodel/hbase/InsertRowTest.java

```diff
@@ -0,0 +1,390 @@
+     */
+    @Test
+    public void testCreatingTheHBaseClientWithRowKeyNull() {
+        try {
+            final HBaseTable table = createHBaseTable(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO, CF_BAR, null);
+            final LinkedHashMap<HBaseColumn, Object> row = createRow(table, HBaseDataContext.FIELD_ID, CF_FOO, CF_BAR,
+                    false);
+            final HBaseColumn[] columns = convertToHBaseColumnsArray(getHBaseColumnsFromRow(row));
+            final Object[] values = new String[] { null };
+            new HBaseClient(getDataContext().getConnection()).insertRow(table.getName(), columns, values, 0);
+            fail("Should get an exception that the indexOfIdColumn is incorrect");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Can't insert a row without having (correct) tableName, columns, values or indexOfIdColumn", e
+                    .getMessage());
+        }
+    }
+
+    /**
+     * Inserting a row without setting enough values directly on the HBaseClient, should throw exception.
+     * NOTE: This exception is already prevented when using the {@link HBaseRowInsertionBuilder}
+     * @throws IOException 
+     */
+    @Test
+    public void testNotSettingEnoughValues() throws IOException {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+            final LinkedHashMap<HBaseColumn, Object> row = createRow(existingTable, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR, false);
+            final List<HBaseColumn> columns = getHBaseColumnsFromRow(row);
+            final Collection<Object> values = getToLittleValues(row);
+            final HBaseClient hBaseClient = ((HBaseDataContext) getUpdateCallback().getDataContext()).getHBaseClient();
+            hBaseClient.insertRow(TABLE_NAME, columns.toArray(new HBaseColumn[columns.size()]), values.toArray(
+                    new Object[values.size()]), 0); // TODO: find the ID-column
+            fail("Should get an exception when insering directly into the HBaseClient without having enough values.");
+        } catch (IllegalArgumentException e) {
+            assertEquals("The amount of columns don't match the amount of values", e.getMessage());
+        }
+    }
+
+    /**
+     * Goodflow. Using an existing table and columns, should work
+     */
+    @Test
+    public void testInsertIntoWithoutExecute() {
+        try {
+            final HBaseTable existingTable = createAndAddTableToDatastore(TABLE_NAME, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR);
+            final LinkedHashMap<HBaseColumn, Object> row = createRow(existingTable, HBaseDataContext.FIELD_ID, CF_FOO,
+                    CF_BAR, false);
+            final List<HBaseColumn> columns = getHBaseColumnsFromRow(row);
+            getUpdateCallback().insertInto(existingTable, columns);
+        } catch (Exception e) {
```

Comment:
Same like earlier, don't catch the exception but have the method throw `IOException` instead.